pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -327,9 +391,14 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 			if err = ctrl.bind(volume, claim); err != nil {
 				// On any error saving the volume or the claim, subsequent
 				// syncClaim will finish the binding.
+				// record count error for provisioning
+				ctrl.recordMetric("provision", claimToClaimKey(claim), err)
```

Comment:
Hmm this is also tricky calling it provision.  It could also be the other two cases:

- Existing PV in same storageclass was found
- Existing PV was created with PV.ClaimRef set

To truly distinguish dynamic provisioning, I think we need to check the "provisioned-by" annotation. But this is also not going to be accurate for local volumes using the static provisioner either :-/ 

Everything else, maybe we should just call "binding" (including prebind).

Reply:
after discussion, there will be four operations:
1. provision, no existing volume found, and claim has volume ref set to ""
2. binding, claim has volume ref set to "", found an existing volume to bind to the claim
3. prebinding, user supplied a volume ref
4. deletion
this has been included into the new commit